Rework SetNameMenu to use ui classes instead of graphics util funcs

Endaris opened the pull request:
Fixes #1074 
I did go some extra steps because the scene was still rather mainloop oriented and using old graphic utils functions instead of the new UI for labels.
I also added a confirmation button so it becomes easier to escape the screen with touch.

Sankyr commented:
nits:
- text hovers at the top of the input box instead of the bottom
- too much space between the "Enter your name" prompt and the text box
  - I see it is partially to include space for the blank name warning (spacing could be made dynamic?)

Endaris commented:
> nits:
> 
>     * text hovers at the top of the input box instead of the bottom
> 
>     * too much space between the "Enter your name" prompt and the text box
>       
>       * I see it is partially to include space for the blank name warning (spacing could be made dynamic?)

I moved the button down a bit so the spacing looks somewhat symmetric while the warning is not there
Text is now aligned.
I figured I was wasting way too much time on this, should have just winged it because we will never have more than 2, maybe 3 inputfields in the entire codebase so reusability is no real concern.
I removed the dynamic cursor placement for now because it's just a nice extra and I couldn't get it working again within a minute and I decided it's worth the time given how rarely inputFields are used.
